nxcomp/configure.ac

```diff
@@ -39,6 +35,43 @@ m4_ifdef([PKG_INSTALLDIR], [PKG_INSTALLDIR],
                        [],[with_pkgconfigdir='$(libdir)/pkgconfig'])
           AC_SUBST([pkgconfigdir], [${with_pkgconfigdir}])])
 
+AC_PATH_PROG(PKGCONF, pkg-config)
```

Comment:
I don't like that...

We should keep using `PKG_CHECK_MODULES(JPEG, libjpeg)` (and actually "convert" this and all the other calls to `PKG_CHECK_MODULES([...], [...])` instead).

`PKG_CHECK_MODULES()` has the following declaration:
> `PKG_CHECK_MODULES(VARIABLE-PREFIX, MODULES, [ACTION-IF-FOUND], [ACTION-IF-NOT-FOUND])`

The default `ACTION-IF-NOT-FOUND` action will break out with an error, but we can also provide a special function to handle that case (i.e., in our case, the code in the `else` branch).

```m4
AC_DEFUN([LIBJPEG_CHECK],[
AC_MSG_CHECKING([for libjpeg shared libary file and headers])
AC_CHECK_LIB([jpeg], [jpeg_destroy_compress],
	[have_jpeg_lib=yes], [have_jpeg_lib=no])
AC_CHECK_HEADERS([jpeglib.h],
	[have_jpeg_headers=yes], [have_jpeg_headers=no])

if test x"$have_jpeg_lib" = "xyes" && test x"$have_jpeg_headers" = "xyes"; then
	AC_MSG_RESULT([yes])
	JPEG_CFLAGS=""
	JPEG_LIBS="-ljpeg"
else
	AC_MSG_RESULT([no])
	AC_MSG_FAILURE([Could not find libjpeg on your system, make sure
the JPEG shared library and header files are installed.])
fi
])

PKG_CHECK_MODULES([JPEG], [libjpeg], [], [LIBJPEG_CHECK])
```

I'd put the `LIBJPEG_CHECK` function in `nx-macros.m4`.

Please test that, though. Both on a system that has pkgconfig files for libjpeg and on a system which doesn't have them. Pay close attention to the output. I don't know if and how nested `AC_MSG_CHECKING`/`AC_MSG_RESULT` work out.